db_dtypes/core.py

```diff
@@ -121,6 +113,16 @@ def _validate_scalar(self, value):
         """
         return self._datetime(value)
 
+    def _validate_setitem_value(self, value):
```

Comment:
Note: Per https://github.com/pandas-dev/pandas/pull/45544#issuecomment-1052809232 this is a required override, and will be documented as such in that PR. We had masked the need for this before with the `__setitem__` override.